### ISSUE TITLE
GET queries include a Content-Type: application/json header

### ISSUE THREAD

**Issue by euirim** [apollographql/apollo-ios#1642]
## Bug report

After using the `useGETForQueries` option, the ApolloClient's requests include `Content-Type: application/json` header that triggers a `"POST body sent invalid JSON"` 400 error. Querying the URL used by the client without a request body and this header results in a successful 200 response. I think this is a bug, but I may be wrong and this could be expected behavior (noob to iOS dev).

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 0.40
- Xcode version: 12.3
- Swift version: 5.3
- Package manager: SPM

## Steps to reproduce

Network class:

```swift
import Foundation
import Apollo

class Network {
    static let shared = Network()
    
    private(set) lazy var apollo: ApolloClient = {
        let client = URLSessionClient()
        let cache = InMemoryNormalizedCache()
        let store = ApolloStore(cache: cache)
        let provider = LegacyInterceptorProvider(client: client, store: store)
        let url = URL(string: "https://XXX")!
        let transport = RequestChainNetworkTransport(
            interceptorProvider: provider,
            endpointURL: url,
            useGETForQueries: true
        )
        return ApolloClient(networkTransport: transport, store: store)
    }()
}
```

Run:

```swift
Network.shared.apollo.fetch(query: ArticlesQuery(page: currentPage, offset: 10)) { result in
    switch result {
        case .success(let graphQLResult):
            print("result", result)
        case .failure(let error):
            print("error", error.localizedDescription)
        }
     }
```

**Comment by designatednerd** [comment]
Interesting! I guess this is technically a bug, we just haven't seen any issues with this previously since even though we're sending the header we aren't actually sending the body, and it appears most servers will ignore that if there is no body. 

What server are you using?

**Comment by euirim** [comment]
Graphene-Python.

**Comment by designatednerd** [comment]
I will fix this the next time I have a little time to poke at bugs (because as I said, this is technically incorrect) but you've already found my suggestion for a workaround :)

**Comment by designatednerd** [comment]
Fix has shipped with `0.41.0`
